### ISSUE TITLE
managed network - service can not connect to different host (service)

### ISSUE THREAD

**Issue by fentas** [rancher/rancher#1987]
hey there,
I have a service utilizing multiple hosts + another service (linked to the first on) placed on a host not used by the first service.
Resulting in that the second service is unable to connect to containers of the first service using ips resolved from the service name. 
But if the container of the second service is placed on any host used by the first on everything works fine.

**Comment by deniseschannon** [comment]
@fentas Have you tried pinging containers between different hosts using the IPs? Have you confirmed that UDP ports 500 and 4500 are open between all hosts?

**Comment by fentas** [comment]
Yes, they share all the same network configuration and udp:500; udp:4500 are open.
The funny thing is that if the second service/container is on one of the hosts of the first service it can connect to all hosts/containers used by the first one. 

Background: I am running in the first service mongod instances on different hosts. The second grabs the container ips with `dig +short <servicename>` and connects to them.

P.s.: Only to recap, I should be able to connect to all containers within the managed network, if linked or not?

**Comment by ebuildy** [comment]
Yep but without links => no DNS. You can still connect by IP adresse, replication are a very good usage of new Rancher metadata service, so you can retrieve easily all containers IP of your service:

http://docs.rancher.com/rancher/metadata-service/

There is an article here : http://rancher.com/deploying-a-python-app-with-mongodb-replicaset-using-rancher-and-rancheros/ but the repl. stuff is done manually ;-( bouhhhhhh

**Comment by vincent99** [comment]
Yes, if you know the `10.42.x.y` address any two containers on the managed network should physically be able to talk to each other.  Only containers that are explictly linked together have DNS entries to tell you those IP addresses.  We had a while ago intended to have a mode to lock down communications with iptables so that communication is *only* allowed if things are linked, but that hasn't been built.  But as @ebuildy points out you can get IPs of other things from metadata now.

**Comment by fentas** [comment]
@vincent99 if what you describe is true then I have a problem. :/ 
Even `ping` is only working on containers (different hosts) within a service or if I `ping` from a container  (to a 'service container') placed on a host used by the service.
What could I do to investigate?

**Comment by fentas** [comment]
Quick update: after some cluster restarts this is what I got for now.
For host registration I use a fqdn. I also started `rancher/agent` on the same host as `rancher/server` but using its local ip for `CATTLE_AGENT_IP`. Leading to that agent was fully recognized **but** every container started on this host could not communicating with his peers/agents container (within managed network). As a side point, local ip should not make any difference as every host is in the same network.
After changing it to the hosts external ip some communication could be achieved.

*Some* because of several things. Sometimes a container shows ready but it can't ping any other container. After some time it can. Also *sometimes* patience is asked on the first contact.
Example:
```log
root@dc31ad55b9da:/# ping 10.42.149.191                                         
PING 10.42.149.191 (10.42.149.191): 48 data bytes                               
56 bytes from 10.42.149.191: icmp_seq=4 ttl=62 time=0.943 ms                    
56 bytes from 10.42.149.191: icmp_seq=5 ttl=62 time=0.828 ms                    
56 bytes from 10.42.149.191: icmp_seq=6 ttl=62 time=1.055 ms                    
56 bytes from 10.42.149.191: icmp_seq=7 ttl=62 time=1.023 ms                    
56 bytes from 10.42.149.191: icmp_seq=8 ttl=62 time=0.893 ms                    
56 bytes from 10.42.149.191: icmp_seq=9 ttl=62 time=0.981 ms                    
^C--- 10.42.149.191 ping statistics ---                                         
10 packets transmitted, 6 packets received, 40% packet loss                     
round-trip min/avg/max/stddev = 0.828/0.954/1.055/0.077 ms                      
root@dc31ad55b9da:/# ping 10.42.42.42                                           
PING 10.42.42.42 (10.42.42.42): 48 data bytes                                   
56 bytes from 10.42.42.42: icmp_seq=4 ttl=62 time=0.939 ms                      
56 bytes from 10.42.42.42: icmp_seq=5 ttl=62 time=1.002 ms                      
^C--- 10.42.42.42 ping statistics ---                                           
6 packets transmitted, 2 packets received, 66% packet loss                      
round-trip min/avg/max/stddev = 0.939/0.970/1.002/0.032 ms
```
This occurs most of the times (not always) but only (as said before) the first packages. Afterwards it seems to be working fine.

Next point could be a separated issue. After restarting host/services quite a few times about 20% of the (service) containers stuck in `Networking` (right after the host/agent comes up). The `Network Agent` shows no problem besides this which I guessing means nothing important.
```log
9/19/2015 9:31:29 PMracoon - IKE keying daemon will not be started as /proc/net/pfkey is not
9/19/2015 9:31:29 PM         available or a suitable 2.6 (or 2.4 with IPSEC backport)
9/19/2015 9:31:29 PM         kernel with af_key.[k]o module installed.
```

I hope this helps you to help me. :)

**Comment by deniseschannon** [comment]
Yes, the delay for when packets go back and forth is noted in this issue: https://github.com/rancher/rancher/issues/278.

@ibuildthecloud would be better to answer the reason the external ip would be needed for when you are using rancher/server and rancher/agent on the same machine. 

For the containers stuck in "Networking", do they eventually get resolved? Or are they permanently stuck in Networking.

**Comment by fentas** [comment]
@deniseschannon After some restart and the latest updates no problems came up yet. I would consider this issue closed for now.
The problem with the two different ips (same host) for rancher/server and rancher/agent could be solved as well? I didn't tried it lately. Maybe creating a new issue for that?

**Comment by deniseschannon** [comment]
@fentas I was able to use a single AWS instance to launch rancher/server and rancher/agent. I used the private IP for my rancher/agent and it had no issues launching containers on it. Though to be fair, I didn't try adding other hosts in the same network to see if it could ping across hosts that were in the same private network but I don't see why it wouldn't work.

If you are still seeing issues, let's reopen another issue with a specific example and issues as this issue is considered closed.
